### ISSUE TITLE
Enchanting Gilded Armor crashed mod

### ISSUE THREAD

**Issue by Scarl37** [scaldings/gilded-netherite-fabric#1]
I was enchanting my armor and I forgot I had Thorns III on the gilded armor already.  When I used the anvil and put the Thorns III enchanted book in the second box, it crashed the game.  As a odd consequence to the crash, it duplicated the Thorns III enchanting book. Attached is the crash report.

[crash-2020-11-29_20.16.35-server.txt](https://github.com/scaldings/gilded-netherite-fabric/files/5612807/crash-2020-11-29_20.16.35-server.txt)

Also tried to enchant a helmet with Thorns III that did not have the enchantment on it and the server crashed again.   Tested in single player mode and had no issues.  Seems only to crash on a server.   The mod is installed on both server and client side.  Here is the file for the second crash.   On this crash the helmet and the enchantment book was destroyed.

[crash-2020-11-29_20.27.28-server.txt](https://github.com/scaldings/gilded-netherite-fabric/files/5612819/crash-2020-11-29_20.27.28-server.txt)

**Comment by scaldings** [comment]
Will take a look at it. Thank you!

**Comment by scaldings** [comment]
Unfortunately, this is an issue with the way I deal with custom materials not having the knockback resistance attribute. I will try to find another method but for now, I will close this issue. I will put this issue on my priority list, but it probably won't be resolved today. If you think you can solve it, just make a pull request. Thank you for reporting this issue!

**Comment by scaldings** [comment]
I have release a new update. Please let me know if the crashes persist.

**Comment by Scarl37** [comment]
I just tested again, and the crash is still persistent.

I tried with an already enchanted piece of armor, and when I try to apply an enchantment to it via the anvil from an enchanted book, it crashed.

Here is the log for that:     [crash-2020-11-30_00.00.43-server.txt](https://github.com/scaldings/gilded-netherite-fabric/files/5613158/crash-2020-11-30_00.00.43-server.txt)

When I tried an non-enchanted piece of armor and tried to enchant it with Unbreaking III via the anvil, it crashed again.  

Here is the log for that:     [crash-2020-11-30_00.03.19-server.txt](https://github.com/scaldings/gilded-netherite-fabric/files/5613159/crash-2020-11-30_00.03.19-server.txt)

For both instances the armor was lost along with the enchanted book.
